src/java.base/share/classes/jdk/internal/classfile/instruction/MonitorInstruction.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.classfile.instruction;
+
+import jdk.internal.classfile.CodeElement;
+import jdk.internal.classfile.CodeModel;
+import jdk.internal.classfile.Instruction;
+import jdk.internal.classfile.Opcode;
+import jdk.internal.classfile.impl.AbstractInstruction;
+import jdk.internal.classfile.impl.Util;
+
+/**
+ * Models a {@code monitorenter} or {@code monitorexit} instruction in the
+ * {@code code} array of a {@code Code} attribute.  Delivered as a {@link
+ * CodeElement} when traversing the elements of a {@link CodeModel}.
+ */
+public sealed interface MonitorInstruction extends Instruction
+        permits AbstractInstruction.UnboundMonitorInstruction {
+
+    /**
+     * {@return a monitor instruction}
+     *
+     * @param op the opcode for the specific type of monitor instruction,
+     *           which must be of kind {@link Opcode.Kind#MONITOR}
+     */
+    static MonitorInstruction of(Opcode op) {
```

Comment:
There are only two cases here - perhaps also offer factories for monitor enter/exit? Or is creating instruction models a rare operation (e.g. because when adapting you always also have a CodeBuilder which has the user-friendly methods?)

Reply:
Each of the approaches server different purposes. `CodeBuilder` with user-friendly methods is my first choice when typing code from scratch, however in transformations you usually start with a pattern switch and it is a Classfile API "standard" to find relevant factory methods in each `Instruction` sub-type.